jmsSource/REAMDE.md

```diff
@@ -0,0 +1,321 @@
+# Overview
+
+The following documentation outlines how to incorporate a JMS Source as part of your project. This allows a user to construct
+applications that interact with a JMS Server. These interactions include the ability to produce messages to and consume 
+messages from JMS Topics and Queues.
+
+In order to incorporate this Extension Source, you will need to set up your local machine with a JMS Server Driver that can be
+used to connect to your JMS Server. Once you have done this, you will need to create the Source in the VANTIQ Modelo IDE. 
+The documentation has been split into two parts, [Setting Up Your Machine](#machine) and [Setting Up Your VANTIQ Modelo IDE](#vantiq).
+
+# Prerequisites <a name="pre" id="pre"></a>
+
+**IMPORTANT:** Read the [Testing](#testing) section before building this project.
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the [Extension Source README.md](../README.md) 
+for more information.
+
+The user must [define the JMS Source implementation](../README.md#-defining-a-typeimplementation) in the VANTIQ Modelo IDE. 
+For an example of the definition, please see the [*jmsImpl.json*](src/test/resources/jmsImpl.json) file located in the 
+*src/test/resources* directory.
+
+Additionally, an example project named *jmsExample.zip* can be found in the *src/test/resources* directory.
+
+*   It should be noted that this example connects to a WebLogic JMS Server.
+
+# Setting Up Your Machine <a name="machine" id="machine"></a>
+
+## Repository Contents
+
+*   **JMSMain** -- The main function for the program. Connects to sources as specified in a configuration file.
+*   **JMSCore** -- Coordinates the communincation between the VANTIQ IDE and the JMS Server.
+*   **JMSHandleConfiguration** -- Sets up the JMS Server connection based on the source's configuration document, and
+    initializes the queryHandler and publishHandler.
+*   **JMS** -- The class that directly interacts with the JMS Server, executing the query and publish requests as sent
+    by the VANTIQ Modelo IDE. Uses classes in the ```io.vantiq.extsrc.jmsSource.communication``` package to produce and 
+    consume messages to or from the JMS Queues and Topics.
+
+## How to Run the Program
+
+1.  In order to effectively use the JMS Source, you will need to download the appropriate JMS Driver with respect to the
+    JMS Server that you are using. Once you have installed this, you will need to create an environment variable named
+    `JMS_DRIVER_LOC` that contains the location of the jar file (*i.e.* `/Users/yourName/somePath/wlthint3client.jar`)
```

Comment:
Once this is in (or nearly so), I'll add something in the overview readme about the new way that the settings.gradle file works.